src/client/utils/tracking/trackOfferEvent.ts

```diff
@@ -100,62 +105,51 @@ export const useTrackOfferEvent = () => {
 
   const [selectedInsuranceTypes] = useSelectedInsuranceTypes()
 
-  const { data: quoteCartQueryData } = useQuoteCartQuery({
-    variables: {
-      id: quoteCartId,
-      locale: isoLocale,
-    },
-  })
-
-  const isReferralCodeUsed =
-    getMonthlyCostDeductionIncentive(quoteCartQueryData) !== undefined
-
-  const selectedBundleVariant = getSelectedBundleVariant(
-    quoteCartQueryData,
-    selectedInsuranceTypes,
-  )
-
-  const [eventQueue, setEventQueue] = useState<EventParameters[]>([])
-
-  const trackOfferCallback = useCallback(
-    (
-      { eventName, options = {} }: EventParameters,
-      selectedBundleVariant: QuoteBundleVariant,
-    ) => {
-      trackOfferEvent(
-        eventName,
-        selectedBundleVariant.bundle,
-        isReferralCodeUsed,
-        {
-          quoteCartId,
-          ...options,
-        },
-      )
-    },
-    [isReferralCodeUsed, quoteCartId],
-  )
-
   const trackOfferHandler = useCallback(
-    (eventParams: EventParameters) => {
-      if (selectedBundleVariant) {
-        trackOfferCallback(eventParams, selectedBundleVariant)
-      } else {
-        setEventQueue((prevQueue) => [...prevQueue, eventParams])
+    ({ eventName, options = {} }: EventParameters) => {
+      const runQuery = async () => {
+        if (apolloClient && quoteCartId) {
+          try {
+            const quoteCartQuery = await apolloClient!.client.query<
+              QuoteCartQuery
+            >({
+              query: QuoteCartDocument,
+              variables: {
+                id: quoteCartId,
+                locale: isoLocale,
+              },
+            })
+            return quoteCartQuery.data
+          } catch (e) {
+            return
+          }
+        } else return
```

Comment:
Guess you could handle this with "finally" block... but shouldn't we do something/log something is the query fails?